src/cbc_sdk/audit_remediation/differential.py

```diff
@@ -0,0 +1,293 @@
+
+        Raises:
+            ApiError: If invalid values are passed in the list.
+        """
+        if not all(isinstance(device_id, int) for device_id in device_ids):
+            raise ApiError("One or more invalid device IDs")
+        self._update_criteria("device_id", device_ids)
+        return self
+
+    def count_only(self, count_only):
+        """
+        Return only count of diff results per device or complete diff metadata result.
+
+        The default value is true, which means only the count will be returned.
+
+        Example:
+            >>> run = cb.select(Differential).newer_run_id(newer_run_id).count_only(True)
+
+        Args:
+            count_only (string): Boolean that indicates whether to return actual metadata
+                or return just the count of differances
+
+        Returns:
+            DifferentialQuery: This instance.
+
+        Raises:
+            ApiError: If invalid values are passed in the list.
+        """
+        if str(count_only).lower() not in ["true", "false"]:
+            raise ApiError("Invalid boolean value for count_only")
+        if str(count_only).lower() == "false":
+            self._count_only = False
+        return self
+
+    def _build_request(self):
+        """
+        Creates the request body for an API call.
+
+        Returns:
+            dict: The complete request body.
+        """
+        request = {"newer_run_id": self._newer_run_id}
+
+        if self._older_run_id:
+            request["older_run_id"] = self._older_run_id
+        if self._criteria:
+            request["criteria"] = self._criteria
+        if self._count_only is False:
+            request["count_only"] = False
+        return request
+
+    def _build_url(self, tail_end=None):
+        """
+        Creates the URL to be used for an API call.
+
+        Args:
+            tail_end (str): String to be appended to the end of the generated URL.
+
+        Returns:
+            str: The complete URL.
+        """
+        url = self._doc_class.urlobject.format(self._cb.credentials.org_key)
+        if tail_end is not None:
+            url += tail_end
+        return url
+
+    def _count(self):
+        """
+        Returns the number of results from the run of this query.
+
+        Returns:
+            int: The number of results from the run of this query.
+        """
+        return self._total_results
+
+    def _perform_query(self, from_row=None, max_rows=None):
+        """
+        Performs the query and returns the results of the query in an iterable fashion.
+
+        Args:
+            from_row (int): Not used; retained for compatibility.
+            max_rows (int): Not used; retained for compatibility.
+
+        Returns:
+            Iterable: The iterated query.
+        """
+        url = self._build_url()
+        request = self._build_request()
+        results = self._cb.post_object(url, body=request).json()
+        result = results.get("results")[0]
```

Comment:
Will we only have one differential in the results? If so we should add a comment explaining this for now in case we need to modify this behavior in the future

Reply:
The result is just one always, but there can be multiple objects in the diff_results. All the differentials for multiple devices are there:
`
{
  "org_key": "4JDT3MX9Q",
  "results": [
    {
      "newer_run_id": "w6zhfxwgr9vujflop0rdqj7krtl09vlr",
      "newer_run_create_time": "2022-09-27T19:13:46.731Z",
      "older_run_id": "7obm6dhnjxuo35ahcmglcpnjpfuoucsb",
      "older_run_create_time": "2022-09-27T18:56:38.133Z",
      "diff_processed_time": 0.049,
      "newer_run_not_responded_devices": [
        17331059
      ],
      "older_run_not_responded_devices": [
        17331059
      ],
      "diff_results": [
        {
          "device_id": 16801738,
          "change_count": 1,
          "added_count": 1,
          "removed_count": 0,
          "changes": [
            {
              "action": "ADDED",
              "fields": [
                {
                  "key": "name",
                  "value": "Facebook Container"
                }
              ]
            }
          ],
          "older_run_row_count": 17,
          "newer_run_row_count": 18
        },
        {
          "device_id": 11412673,
          "change_count": 2,
          "added_count": 2,
          "removed_count": 0,
          "changes": [
            {
              "action": "ADDED",
              "fields": [
                {
                  "key": "name",
                  "value": "Disconnect"
                }
              ]
            },
            {
              "action": "ADDED",
              "fields": [
                {
                  "key": "name",
                  "value": "HTTPS Everywhere"
                }
              ]
            }
          ],
          "older_run_row_count": 22,
          "newer_run_row_count": 24
        }
      ]
    }
  ]
}
`